tripal/tripal
# Tripal 4 ChadoContactDefault field

# Tripal 4 Core Dev Task

## Issue #1414 and #1550

### Tripal Version: 4.alpha1-dev

## Description

This is the initial implementation of the ChadoContactDefault field.

There are only two tables with a `contact_id` column that will be able to use this field:
  * `stockcollection` (but is not a default content type)
  * `study`

and in addition the `arraydesign` table has a `manufacturer_id` which maps to the contact table

Note that before this PR, we can't create a study because `contact_id` has a NOT NULL constraint in the study table. This PR includes an update to the YAML as described in issue #1550 so that it is easy to test.

## Dependency
As of https://github.com/tripal/tripal/pull/1599/commits/591868140958fe6776e885856bd8cf979956e629, this PR is dependent on pull #1604. ~It cannot be merged until that one is merged.~ it has been merged

## Testing?
I will test on both the Study and Array Design content types. Currently you can't create a study because the `contact_id` is NOT NULL. This new field will fix that! Similar problem for Array Design.

1. Create two contact records, one a `Person (TCONTACT:0000003)` and one a `Company (NCIT:C54131)` (though types do not really matter)
![contacts](https://github.com/tripal/tripal/assets/8419404/201a5eca-94b8-45e3-8619-c0eb74381202)

2. Now create a study, and specify the person contact.
![study1](https://github.com/tripal/tripal/assets/8419404/24966696-8296-4737-8cf4-e2e0861c7265)

3. It should appear similar to this
![study2](https://github.com/tripal/tripal/assets/8419404/7056f900-3bc0-42cb-b5f5-7af19ff52fc8)

4. Create an array design, and specify the company contact as the "Manufacturer". "Platform Type" is also a required field, just enter any random CV term there, I used "DNA".
![ad1](https://github.com/tripal/tripal/assets/8419404/c81c1ca0-86d4-458f-8f8a-181c22328e36)

5. Content should be saved and appear something like this
![ad2](https://github.com/tripal/tripal/assets/8419404/8e5b7c34-0b21-4b58-a9dc-f54ddd45e835)

If you happen to notice that that "Platform Type" is blank, that is actually issue #1398 but not relevant to this PR
